harness_tests/pex/main1.c

```diff
@@ -0,0 +1,113 @@
+#include <test/test.h>
+#include <pex/pex.h>
+
+/*
+    The goals of this harness test are:
+    1. Make sure pex initializes correctly
+    2. We can write to entire registers
+    3. We can read individual pins
+    4. We can set individual pins
+*/
+
+pin_info_t cs = {
+    .port = &PEX_CS_PORT_PAY,
+    .ddr = &PEX_CS_DDR_PAY,
+    .pin = PEX_CS_PIN_PAY
+};
+
+pin_info_t rst = {
+    .port = &PEX_RST_PORT_PAY,
+    .ddr = &PEX_RST_DDR_PAY,
+    .pin = PEX_RST_PIN_PAY
+};
+
+pex_t pex = {
+    .addr = PEX_ADDR_PAY,
+    .cs = &cs,
+    .rst = &rst
+};
+
+// Test that the pex has been correctly initialized
+void init_pex_test(void) {
+    init_pex(&pex);
+
+    ASSERT_EQ(read_register(&pex, PEX_IOCON), PEX_IOCON_DEFAULT);
+
+    // Check that the IO register pins, aka the direction, default to input (1)
+    ASSERT_EQ(read_register(&pex, PEX_IODIR_A), 0xEE);
```

Comment:
Should this be 0xFF (if it's supposed to be all 1s)? Check the datasheet

Reply:
Oh yeah, forgot 1111 was F